test/index.js

```diff
@@ -1338,3 +1338,22 @@ describe('parse', function() {
   });
 
 });
+
+describe('regex', function() {
+
+  describe('match', function() {
+
+    it('returns a Just if there is a match', function() {
+      eq(S.match(/(good)?bye/, 'goodbye').toString(), 'Just(["goodbye", "good"])');
+      eq(S.match(/([a-z]a)/g, 'bananas').toString(), 'Just(["ba", "na", "na"])');
```

Comment:
Nice pattern, though the capturing group is unnecessary. Capturing groups are ignored when the `g` flag is present.

It would be nice to separate these tests into different groups:

```javascript
it('returns a Just if there is a match', ...);

it('returns a Nothing() if no match', ...);

it('supports global patterns', ...);

it('supports (optional) capturing groups', ...);
```